### PR TITLE
ALTAPPS-795: iOS streak recovery bottom sheet

### PR DESCRIPTION
**Pull request by vladkash**
**YouTrack Issues**:
[#ALTAPPS-795](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-795)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**
- Added streak recovery modal on iOS

### CODE REVIEW

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/Streak/Views/Modals/StreakRecoveryModalView.swift [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                .buttonStyle(OutlineButtonStyle())
+            }
+        }
+        .padding([.horizontal, .bottom])
+        .onAppear {
+            streakRecoveryModalDelegate?.streakRecoveryModalDidAppear()
+        }
+        .onDisappear {
+            streakRecoveryModalDelegate?.streakRecoveryModalDidDisappear()
+        }
```
Comment (ivan-magda) [inline iosHyperskillApp/iosHyperskillApp/Sources/Modules/Streak/Views/Modals/StreakRecoveryModalView.swift:73]: Let's don't use SwiftUI callbacks, they work not as we expected. That's why we have `UIViewControllerEventsWrapper`.

I propose to move this to the `StreakRecoveryModalViewController`.
See for example `StageImplementUnsupportedModalViewController` how we are handling those callbacks.

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/Streak/Views/Modals/StreakRecoveryModalDelegate.swift
Comment (ivan-magda) [inline iosHyperskillApp/iosHyperskillApp/Sources/Modules/Streak/Views/Modals/StreakRecoveryModalDelegate.swift:1]: 1. Let's move this delegate to the `StreakRecoveryModalViewController` file and name it `StreakRecoveryModalViewControllerDelegate`.
2. Delegate function name should start with the name of the controller, for example, `func streakRecoveryModalViewControllerDidAppear`, https://vyahhi.myjetbrains.com/youtrack/articles/ALTAPPS-A-30/Swift-Style-Guide#delegaty
3. Remove `streakRecoveryModalDelegate` from the `StreakRecoveryModalView`, view will notify the controller about button clicks via callbacks

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/Streak/Views/Modals/StreakRecoveryModalViewController.swift
Comment (ivan-magda) [inline iosHyperskillApp/iosHyperskillApp/Sources/Modules/Streak/Views/Modals/StreakRecoveryModalViewController.swift:1]: I expecting here smth like that:
```swift
import SwiftUI

protocol StreakRecoveryModalViewControllerDelegate: AnyObject {
    ...
}

class StreakRecoveryModalViewController: PanModalSwiftUIViewController<StreakRecoveryModalView> {
    private weak var delegate: StreakRecoveryModalViewControllerDelegate?

    convenience init(
        recoveryPriceAmount: String,
        recoveryPriceLabel: String,
        modalText: String,
        delegate: StreakRecoveryModalViewControllerDelegate?
    ) {
        let view = StreakRecoveryModalView(
            recoveryPriceAmount: recoveryPriceAmount,
            recoveryPriceLabel: recoveryPriceLabel,
            modalText: modalText,
            ...
        )

        self.init(
            isPresented: .init(get: { false }, set: { _ in }),
            content: { view }
        )
        self.delegate = delegate
    }
}
```

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/Streak/Views/Modals/StreakRecoveryModalView.swift [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                        .frame(widthHeight: appearance.gemsBadgeWidthHeight)
+
+                    Text(recoveryPriceAmount)
+                        .foregroundColor(.primaryText)
+                        .font(.body)
+                        .bold()
+
+                    Text(recoveryPriceLabel)
+                        .foregroundColor(.primaryText)
+                        .font(.body)
```
[the remark this answers is not in the document]
Comment (vladkash) [reply iosHyperskillApp/iosHyperskillApp/Sources/Modules/Streak/Views/Modals/StreakRecoveryModalView.swift:47]: 🔥

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/Streak/Views/Modals/StreakRecoveryModalDelegate.swift
Comment (vladkash) [reply iosHyperskillApp/iosHyperskillApp/Sources/Modules/Streak/Views/Modals/StreakRecoveryModalDelegate.swift:1]: Thank you for detailed explanation!

**Review by ivan-magda — APPROVED** [review]

**Comment by ivan-magda** [comment]
@vladkash when merging to the release branch please deploy both platforms, because previously we have incorrect show/hidden events https://github.com/hyperskill/mobile-app/pull/527/commits/720d82fc5b19705a92f6cbedbf735397a52e7ace

**Pull request merged**